gnat2goto/driver/gnat_utils.adb

```diff
@@ -11,6 +11,9 @@ package body GNAT_Utils is
       Nam : constant Node_Id := Name (N);
 
    begin
+      if Nkind (Nam) = N_Explicit_Dereference then
```

Comment:
Because the case expression always returns an `Entity` of something. In the case of `explicit-dereference` the entity-of-prefix-of-name-of-n leads to the return type of the function (while we need the whole type of the function).

Reply:
It might be cleaner to move this into the case expression, and move the `Entity` call from outside the case to the cases that need it

Reply:
Done.